### PR TITLE
riscv: fix register usage

### LINKED ISSUE #9888: RISC-V code generator: inconsistent use of register t0
Register `t0` is allocatable, yet it is not properly tracked by the GC:
https://github.com/ocaml/ocaml/blob/trunk/runtime/riscv.S#L72-L94
`t0` is not saved and restored to/from the `caml_gc_regs` array.  If a pointer to the heap is allocated in `t0` and is live across an allocation point, something wrong will happen at GC time.

On the other hand, according to some comments, `t0` should not be allocated as it is intended to be used as a temporary:
https://github.com/ocaml/ocaml/blob/trunk/asmcomp/riscv/proc.ml#L58-L59

Please clarify the intended use of `t0`.

### PR DESCRIPTION
**Pull request by nojb**
Fixes #9888

### CODE REVIEW

File: asmcomp/riscv/proc.ml [hunk 1]
```diff
@@ -55,7 +56,7 @@ let word_addressed = false
   Additional notes
   ----------------
 
-    - t0-t1 are used by the assembler and code generator, so
+    - t1 is used by the assembler and code generator, so
```
Comment (xavierleroy) [inline asmcomp/riscv/proc.ml]: Is it really "used by the assembler"?  The MIPS and Alpha assemblers silently used a temporary register for some macro-instructions.  I don't think this is the case for the RISC-V assembler.

**Review by xavierleroy — COMMENTED** [review]
I believe this fixes #9888.  One question below prompted by a comment.

File: asmcomp/riscv/proc.ml (hunk 1 above)
Comment (nojb) [reply asmcomp/riscv/proc.ml]: Indeed you are right. the original comment meant that it was used in `riscv.S`, but it is confusing. I amended the comment accordingly.

**Review by xavierleroy — APPROVED** [review]
Looks good to me, thanks!

**Comment by nojb** [comment]
Merged, thanks!

**Pull request merged**